pkgs/stdenv/linux/bootstrap-tools/scripts/unpack-bootstrap-tools.sh

```diff
@@ -21,18 +21,29 @@ fi
 # use a copy of patchelf.
 LD_LIBRARY_PATH=$out/lib $LD_BINARY $out/bin/cp $out/bin/patchelf .
 
+# make a copy of /lib so we don't try patchelf'ing a library that
+# patchelf itself links against -- this may cause segfaults due to
+# lazy loading of binary images from disk
+LD_LIBRARY_PATH=$out/lib $LD_BINARY $out/bin/cp -r $out/lib lib
+
 for i in $out/bin/* $out/libexec/gcc/*/*/*; do
     if [ -L "$i" ]; then continue; fi
     if [ -z "${i##*/liblto*}" ]; then continue; fi
     echo patching "$i"
-    LD_LIBRARY_PATH=$out/lib $LD_BINARY \
+    LD_LIBRARY_PATH=lib $LD_BINARY \
         ./patchelf --set-interpreter $LD_BINARY --set-rpath $out/lib --force-rpath "$i"
 done
 
-for i in $out/lib/librt-*.so $out/lib/libpcre*; do
+for i in $out/lib/lib*.so*; do
     if [ -L "$i" ]; then continue; fi
+    # do not --set-rpath libgcc_s.so.1, because doing so will
+    # contaminate stdenv with forbidden references to bootstrap-tools
+    if [ "x$i" == "x$out/lib/libgcc_s.so.1" ]; then continue; fi
+    # do not --set-rpath if the library had no RPATH in the first place
```

Comment:
Thanks, I will update this in the morning.

Reply:
> Thanks, I will update this in the morning.

Sorry I am running behind here; I finally sorted out #168983 which took way longer than I thought it would.  I will get back to this PR tomorrow or Thursday.

Reply:
Incorporated in latest push.